### PR TITLE
ci: various tweaks

### PR DESCRIPTION
**Pull request by Kichura**
This updates the GitHub Actions to make use of the new operating systems and other new dependencies, Do note that this requires testing before merging.

### CODE REVIEW

**Comment by vgskye** [comment]
I think older Ubuntu is preferable due to glibc version

**Comment by Kichura** [comment]
> I think older Ubuntu is preferable due to glibc version

And why exactly?

**Comment by vgskye** [comment]
apps built on older glibc will run on newer glibc but not the other way I believe

**Comment by Kichura** [comment]
> apps built on older glibc will run on newer glibc but not the other way I believe

I could try to deploy older glibc to the distribution and yolo it cuz downgrading back to old ubuntu release isn't recommended for specific reasons. *(mainly support is the priority)*

**Comment by Kichura** [comment]
Software can be tested now. *(See github actions list of own fork)*

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -13,13 +13,13 @@ jobs:
...
     steps:
-      - name: Checkout the Repository
+      - name: Checkout Repository
         uses: actions/checkout@v3
       - name: Setup Node
         uses: actions/setup-node@v3
         with:
           node-version: 18
-      - name: Setup pnpm
+      - name: Setup pNpm
```
Comment (vgskye) [inline .github/workflows/build.yml]: nononononono what the fuck is this capitalization

Comment (Kichura) [reply .github/workflows/build.yml]: This was what it was mentioned as prior, will revert.

**Pull request merged**